ArduCopter/mode_guided.cpp

```diff
@@ -286,47 +315,89 @@ bool ModeGuided::set_destination(const Location& dest_loc, bool use_yaw, float y
         pos_control_start();
     }
 
-    if (!wp_nav->set_wp_destination_loc(dest_loc)) {
-        // failure to set destination can only be because of missing terrain data
-        AP::logger().Write_Error(LogErrorSubsystem::NAVIGATION, LogErrorCode::FAILED_TO_SET_DESTINATION);
-        // failure is propagated to GCS with NAK
-        return false;
-    }
-
     // set yaw state
     set_yaw_state(use_yaw, yaw_cd, use_yaw_rate, yaw_rate_cds, relative_yaw);
 
+    // set position target and zero velocity and acceleration
+    Vector3f pos_target_f;
+    if (!dest_loc.get_vector_from_origin_NEU(pos_target_f)) {
+        guided_pos_target_cm = pos_control->get_pos_target_cm();
+    } else {
+        guided_pos_target_cm = pos_target_f.topostype();
+    }
+
+    guided_vel_target_cms.zero();
+    guided_accel_target_cmss.zero();
+
     // log target
-    copter.Log_Write_GuidedTarget(guided_mode, Vector3f(dest_loc.lat, dest_loc.lng, dest_loc.alt),Vector3f());
+    copter.Log_Write_GuidedTarget(guided_mode, Vector3f(dest_loc.lat, dest_loc.lng, dest_loc.alt), guided_vel_target_cms, guided_accel_target_cmss);
 
     send_notification = true;
 
     return true;
 }
 
-// guided_set_velocity - sets guided mode's target velocity
+// set_velaccel - sets guided mode's target velocity and acceleration
```

Comment:
nitpick: tiny comment issue in that the comments don't match the method.  I guess this method only sets the accel targets.